### PR TITLE
feat(update_replication_factor#1): support to get the replication factor of each table

### PR DESCRIPTION
**Pull request by empiredan**
https://github.com/apache/incubator-pegasus/issues/865

### CODE REVIEW

**Review by levy5307 — APPROVED** [review]
LGTM

**Comment by hycdong** [comment]
I have a suggestion about your unit test. I noticed that you updated code in `src/meta/test/main.cpp`, `src/meta/test/meta_service_test_app.h` and `src/meta/test/server_state_test.cpp`. In `src/meta/test/server_state_test.cpp`, you added some rpc logic to send and receive the rpc request and response. Actually, in current meta unit test implemetation, this logic can be reused.
I recommend you adding your unit test in `meta_app_opeation_test.cpp`, the rpc logic is hidden through `meta_test_base.cpp`, your ut code can be strongly simplified.
Besides, you don't need to update `src/meta/test/main.cpp` and `src/meta/test/meta_service_test_app.h`.

File: src/meta/test/server_state_test.cpp [hunk 1]
```diff
@@ -1,8 +1,15 @@
 #include <gtest/gtest.h>
+
```
Comment (hycdong) [inline src/meta/test/server_state_test.cpp:2]: As the comments shows, I recommend you adding your unit test in `src/meta/test/meta_app_operation_test.cpp`.

File: src/meta/test/meta_app_operation_test.cpp [hunk 2]
```diff
@@ -104,6 +105,31 @@ class meta_app_operation_test : public meta_test_base
...
+        return rpc.response();
+    }
+
+    void set_partition_max_replica_count(const std::string &app_name,
+                                         int32_t partition_index,
+                                         int32_t max_replica_count)
+    {
+        dsn::zauto_write_lock l(_ss->_lock);
+
+        auto app = _ss->get_app(app_name);
```
Comment (hycdong) [inline src/meta/test/meta_app_operation_test.cpp:126]: L124- L126 can be simplified, you can use function `find_app` in `meta_test_base.cpp`

File: src/meta/test/meta_app_operation_test.cpp [hunk 3]
```diff
@@ -22,6 +22,7 @@
 #include "meta_service_test_app.h"
 #include "meta_test_base.h"
 #include "meta/meta_split_service.h"
+#include "misc/misc.h"
```
Comment (hycdong) [inline src/meta/test/meta_app_operation_test.cpp:25]: Why would you add this header file?

Comment (empiredan) [reply src/meta/test/meta_app_operation_test.cpp:25]: `misc/misc.h` is included to use `random32` to choose a random partition in L431 ~

**Review by hycdong — APPROVED** [review]

**Review by levy5307 — APPROVED** [review]

**Pull request merged**